packages/doenetml-worker-rust/lib-doenetml-core/src/new_core/graph_based_core.rs

```diff
@@ -208,4 +526,191 @@ impl Core {
             }
         }
     }
+
+    /// Add component `child_idx` as the first child of `parent_idx`.
+    fn prepend_child_to_structure_graph(
+        &mut self,
+        parent_idx: ComponentIdx,
+        child_idx: ComponentIdx,
+    ) {
+        let parent_node = GraphNode::Component(parent_idx);
```

Comment:
I'd prefer to force the caller to pass in a `GraphNode::Component(...)` to emphasize that they know what they're doing.